pmlc/rt/vulkan/wrappers.cc

```diff
@@ -168,6 +168,15 @@ void bindBufferFloat32(void *vkRuntimeManager, DescriptorSetIndex setIndex,
   case 3:
     bindBuffer<float, 3>(vkRuntimeManager, setIndex, bindIndex, ptr);
     break;
+  case 4:
+    bindBuffer<float, 4>(vkRuntimeManager, setIndex, bindIndex, ptr);
+    break;
+  case 5:
+    bindBuffer<float, 5>(vkRuntimeManager, setIndex, bindIndex, ptr);
+    break;
+  case 6:
+    bindBuffer<float, 6>(vkRuntimeManager, setIndex, bindIndex, ptr);
+    break;
```

Comment:
Can we cast to an unranked memref so that we don't need all these variations?

Reply:
Actually the static memref has been casted to unranked memref. Every unranked memref is lowered into two variables (int rank, void ptr) after llvm lowering pass. The thing is that here I call a template function(bindbuffer), which has to know the rank at compilation time, but the rank is a variable not a constant(so I use switch case here). I will check if there is a better way to do this.